File: solutions/erlang/space-age/1/src/space_age.erl
-module(space_age).

-export([age/2]).


age(_Planet, _Seconds) -> 
    _Seconds / earth_seconds_a_year() / orbital_period(_Planet).


earth_seconds_a_year() -> 31557600.

orbital_period(mercury) -> 0.2408467 ;
orbital_period(venus) -> 0.61519726 ;
orbital_period(earth) -> 1 ;
orbital_period(mars) -> 1.8808158 ;
orbital_period(jupiter) -> 11.862615 ;
orbital_period(saturn) -> 29.447498 ;
orbital_period(uranus) -> 84.016846 ;
orbital_period(neptune) -> 164.79132 .

File: solutions/go/space-age/1/space_age.go
// Package space calculates measures on other planets.
package space

type Planet string

// Age converts ages given in seconds into years of different planets.
func Age(secs float64, planet Planet) float64 {
	earth := secs / 31557600
	switch planet {
	case "Earth":
		return earth
	case "Mercury":
		return earth / 0.2408467
	case "Venus":
		return earth / 0.61519726
	case "Mars":
		return earth / 1.8808158
	case "Jupiter":
		return earth / 11.862615
	case "Saturn":
		return earth / 29.447498
	case "Uranus":
		return earth / 84.016846
	case "Neptune":
		return earth / 164.79132
	}
	return 0
}